docs/docs/dev_docs/tutorials/token_portal/setup.md

```diff
@@ -0,0 +1,185 @@
+noirup -v 0.12.0-aztec.0
+```
+
+# Create a yarn project
+
+Our root yarn project will house everything ✨
+
+```bash
+mkdir aztec-token-portal && cd aztec-token-portal && yarn init
+```
+
+In your `package.json` put this
+
+```json
+"type": "module",
+  "scripts": {
+    "build": "yarn clean && tsc -b",
+    "build:dev": "tsc -b --watch",
+    "clean": "rm -rf ./dest tsconfig.tsbuildinfo",
+    "start": "yarn build && node ./dest/src/index.js"
+  },
+```
+
+Create a `tsconfig.json` and paste this into it:
+
+```json
+{
+  "compilerOptions": {
+    "rootDir": "./packages",
+    "outDir": "./dest",
+    "target": "es2020",
+    "lib": ["dom", "esnext", "es2017.object"],
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
+    "strict": true,
+    "declaration": true,
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "downlevelIteration": true,
+    "inlineSourceMap": true,
+    "declarationMap": true,
+    "importHelpers": true,
+    "resolveJsonModule": true,
+    "composite": true,
+    "skipLibCheck": true
+  },
+  "include": [
+    "packages/src/**/*",
+    "contracts/**/*.json",
+    "packages/src/**/*",
+    "packages/aztec-contracts/token_bridge/target/*.json"
+  ],
+  "exclude": ["node_modules", "**/*.spec.ts", "contracts/**/*.ts"],
+  "references": []
+}
+```
+
+# Create a nargo project
+
+Now inside `aztec-token-portal` create a new directory called `aztec-contracts`
+
+Inside `aztec-contracts`, create a nargo contract project by running
+
+```bash
+nargo new -contract token_bridge
+```
+
+Your structure will look something like this
+
+```
+aztec-contracts
+└── token_bridge
+    ├── Nargo.toml
+    ├── src
+				├── main.nr
+```
+
+Inside `Nargo.toml` you will need to add some dependencies. Put this at the bottom:
+
+```json
+[dependencies]
+aztec = { git="https://github.com/AztecProtocol/aztec-packages/", tag="aztec-packages-v0.8.7", directory="yarn-project/aztec-nr/aztec" }
+value_note = { git="https://github.com/AztecProtocol/aztec-packages/", tag="aztec-packages-v0.8.7", directory="yarn-project/aztec-nr/value-note"}
+safe_math = { git="https://github.com/AztecProtocol/aztec-packages/", tag="aztec-packages-v0.8.7", directory="yarn-project/aztec-nr/safe-math"}
+```
+
+We will also need some utility functions that we won’t write in this tutorial. In the `src` directory, create a new file called `token_interface.nr` and paste the contents of [this file](https://github.com/AztecProtocol/aztec-packages/blob/891c1362160693f69bc6a843c7c41776b51b9f7c/yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/token_interface.nr#L4).
+
+Then in the same place create a new file called `[util.nr](http://util.nr)` and paste the contents of [this file.](https://github.com/AztecProtocol/aztec-packages/blob/891c1362160693f69bc6a843c7c41776b51b9f7c/yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/util.nr)
+
```

Comment:
Show how the directory looks like here so slow-brain people like me know where how the code exactly looks